### ISSUE TITLE
CiscoSMB false positive diffs / notifications based on uptime

### ISSUE THREAD

**Issue by quacktacular** [ytti/oxidized#1398]
Hey! Updated oxidized today (0.24.0) and I now get notifications every time the check runs.

Looks like its not filtering out uptime:
```
-!  1    175,18:24:38   
-!  2    175,18:27:43   
+!  1    175,18:26:21   
+!  2    175,18:29:26   
```
How can I sort it out?

**Comment by quacktacular** [comment]
This is still an issue on multi unit swithces:
```
! Unit     Up time     
! ---- --------------- 
-!  1    175,20:32:50   
-!  2    175,20:35:24   
-!  3    175,20:37:30   
-!  4    175,20:36:58   
+!  1    175,20:45:31   
+!  2    175,20:48:04   
+!  3    175,20:50:10   
+!  4    175,20:49:38   
```

**Comment by quacktacular** [comment]
@wk also still include temperature information. So it will also notify / diff on temp change.
```
! Unit   Temperature (Celsius)   Temperature Sensor Status 
! ---- ------------------------- ------------------------- 
!  1              47                        OK             
!  2                                        N/A            
!  3              46                        OK             
!  4                                        N/A            
! 
! 
! Unit     Up time     
! ---- --------------- 
!  1    175,19:25:32   
!  2    175,19:28:05   
!  3    175,19:30:11   
!  4    175,19:29:39   
```
I think the "show system" is unnecessary. It doesn't provide much utility (for backups anyway), and it wasn't included before 0.22.0.

**Comment by wk** [comment]
Based on your included output extract, it's apparent this isn't the style output that the regexp has been written to catch, as it is dissimilar from the output of devices provided by users based on which it was written. 

For system uptime, in #1364, the output looks like so:
`!  System Up Time:          49 days, 20 hours, 57 minutes, and 4.47 seconds`

For temperature, in #1394, the output looks like so:
```
!
! System Temperature:
! Unit 1
! Temperature 1: 35 degrees
```

The output above appears to be tabulated and differently styled (as opposed to my initial guess at a partial match). Could you specify the exact model and configuration on which you observe this behavior via e.g. a full output of `show version` and `show system`?

**Comment by wk** [comment]
Earlier reply contained incorrect information based on conflation of two similar but unrelated issues (now removed).

Re-opening this. The `show system` was added [here](https://github.com/ytti/oxidized/commit/2726fff67ded963e24776e60f1e8976ae2bb037b#diff-02d7496a8deb210108de6c71cbc4f729) by @davama, @davama - could you let us know why was this necessary, and which model of these devices do you make use of? @quacktacular could you also include a full output of `show version` and `show system` or similar to understand which devices display their output differently?

**Comment by quacktacular** [comment]
For sure. Here's `show version` from one of my multi-unit switches
```
sw30-me-ca#show version

       Unit             SW version         Boot version         HW version      
------------------- ------------------- ------------------- ------------------- 
         1                1.4.8.6            1.4.0.02               V05         
         2                1.4.8.6            1.4.0.02               V05      
```
and `show system` from the same switch.
```
sw30-me-ca#show system

Unit          Type          
---- ---------------------- 
 1         SG500X-48P       
 2         SG500X-48P       

Unit   FAN       
       Status    
----   --------- 
 1        OK       
 2        OK       

Unit   Temperature (Celsius)   Temperature Sensor Status 
---- ------------------------- ------------------------- 
 1              44                        OK             
 2              42                        OK             


Unit     Up time     
---- --------------- 
 1    176,04:35:07   
 2    176,04:38:13  
```

**Comment by wk** [comment]
Simply removing `show system`, while not ideal due to loss of device model information, seems a lot easier than parsing and scrubbing elements of variable length tabular data correctly. Let's see what @davama would like to do here for stacking support.

**Comment by ytti** [comment]
```
~ ❱ cat str
       Unit             SW version         Boot version         HW version
------------------- ------------------- ------------------- -------------------
         1                1.4.8.6            1.4.0.02               V05
         2                1.4.8.6            1.4.0.02               V05
and show system from the same switch.

sw30-me-ca#show system

Unit          Type
---- ----------------------
 1         SG500X-48P
 2         SG500X-48P

Unit   FAN
       Status
----   ---------
 1        OK
 2        OK

Unit   Temperature (Celsius)   Temperature Sensor Status
---- ------------------------- -------------------------
 1              44                        OK
 2              42                        OK


Unit     Up time
---- ---------------
 1    176,04:35:07
 2    176,04:38:13

~ ❱ ruby -e 'ary=File.read("str").lines;ary=ary[0..ary.find_index{|e|e.match /Up time/}-3];i=ary.find_index{|e|e.match /Celsius/};ary[i+2
    ..-1]=ary[i+2..-1].map{|e|e=e.split(/(\s+)/);e[4]="NA";e.join};puts ary'
       Unit             SW version         Boot version         HW version
------------------- ------------------- ------------------- -------------------
         1                1.4.8.6            1.4.0.02               V05
         2                1.4.8.6            1.4.0.02               V05
and show system from the same switch.

sw30-me-ca#show system

Unit          Type
---- ----------------------
 1         SG500X-48P
 2         SG500X-48P

Unit   FAN
       Status
----   ---------
 1        OK
 2        OK

Unit   Temperature (Celsius)   Temperature Sensor Status
---- ------------------------- -------------------------
 1              NA                        OK
 2              NA                        OK
~ ❱
```

**Comment by wk** [comment]
@davama @quacktacular it would also be helpful if you could include a `show system` output from a unit that is *not* a member of a stack, for comparison.

**Comment by davama** [comment]
@wk 
```
H-TXA-2WDM-SW-1#show system 
System Description:                       SF302-08P 8-Port 10/100 PoE Managed Switch
System Up Time (days,hour:min:sec):       46,23:55:47
System Contact:                           CDE
System Name:                              H-TXA-2WDM-SW-1
System Location:                          FTTX
System MAC Address:                       84:78:ac:a4:cc:79
System Object ID:                         1.3.6.1.4.1.9.6.1.82.8.2

Fans Status:                              N/A
```

Does this help? 

As to why was it added? 
Was added to have hardware model info

**Comment by quacktacular** [comment]
The info is neat, but I think we’d still be better of just getting rid of the show system. If people really wanted it they could override the model.

**Comment by davama** [comment]
@quacktacular agreed

**Comment by bbrendon** [comment]
Oxidized in my environment no longer works because there are too many commits for it to handle. I can only use it from the command line. It appears the problem is because the time changes.

```
# git show c0fc6fff250c8d0363c114fce59df515e57adf42
commit c0fc6fff250c8d0363c114fce59df515e57adf42
Author: IT <it-admin@<redacted>>
Date:   Tue Nov 20 23:24:35 2018

    update sw8.ad.<redacted>

diff --git a/sw8.<redacted> b/sw8.<redacted>
index 4cef43f..59bd405 100644
--- a/sw8.<redacted>
+++ b/sw8.<redacted>
@@ -2,7 +2,7 @@
 ! Boot version    1.3.5.06 ( date  21-Jul-2013 time  15:12:10 )
 ! HW version    V02
 ! System Description:                       SG300-52P 52-Port Gigabit PoE Managed Switch
-! System Up Time (days,hour:min:sec):       184,04:11:28
+! System Up Time (days,hour:min:sec):       184,05:11:51
 ! System Contact:                           <redacted>
 ! System Name:                              sw8
 ! System Location:                          server rm

```

**Comment by wk** [comment]
Should be resolved by #1799 by removing `sh system` outright.
